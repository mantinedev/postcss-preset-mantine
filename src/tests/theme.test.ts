import { testTransform } from './utils';

const baseInput = `
.demo {
  background: theme(color-orange-4);
  border: rem(1px) solid theme(color-gray-filled);
  padding: theme(spacing-md) theme(spacing-sm);
}
`;

const lightDarkInput = `
.demo {
  background: light-dark(theme(color-red-5), theme(color-blue-5));
}
`;

describe('theme', () => {
  it('correctly replaces theme function', async () => {
    const res = await testTransform(baseInput);
    expect(res.css).toMatchSnapshot();
  });

  it('correctly replaces theme with light-dark function', async () => {
    const res = await testTransform(lightDarkInput);
    expect(res.css).toMatchSnapshot();
  });
});
