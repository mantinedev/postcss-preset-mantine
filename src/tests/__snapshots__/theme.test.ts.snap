// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme correctly replaces theme function 1`] = `
"
.demo {
  background: var(--mantine-color-orange-4);
  border: calc(0.0625rem * var(--mantine-scale)) solid var(--mantine-color-gray-filled);
  padding: var(--mantine-spacing-md) var(--mantine-spacing-sm);
}
"
`;

exports[`theme correctly replaces theme with light-dark function 1`] = `
"
.demo {
  background: var(--mantine-color-red-5);
}
  [data-mantine-color-scheme='dark'] .demo {
  background: var(--mantine-color-blue-5);
}
"
`;
